domains/transfer/Transfer.API/API/Claiming/Api/Controllers/ClaimAutomationController.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Threading.Tasks;
+using API.Claiming.Api.Dto.Response;
+using API.Claiming.Api.Models;
+using API.Claiming.Api.Repositories;
+using API.Shared.Extensions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Claiming.Api.Controllers;
+
+[Authorize]
+[ApiController]
+[Route("api/claim-automation")]
+public class ClaimAutomationController : ControllerBase
+{
+    private readonly IClaimAutomationRepository claimAutomationRepository;
+
+    public ClaimAutomationController(IClaimAutomationRepository claimAutomationRepository)
+    {
+        this.claimAutomationRepository = claimAutomationRepository;
+    }
+
+    [HttpPost("start")]
+    [ProducesResponseType(typeof(ClaimAutomationArgumentDto), 201)]
+    [ProducesResponseType(typeof(ClaimAutomationArgumentDto), 200)]
+    public async Task<ActionResult> StartClaimAutomation()
+    {
+        var subject = User.FindSubjectGuidClaim();
+        var claim = await claimAutomationRepository.GetClaimSubject(Guid.Parse(subject));
+        if (claim != null)
+        {
+            var claimSubjectDto = new ClaimAutomationArgumentDto(claim.CreatedAt);
+
+            return Ok(claimSubjectDto);
+        }
+
+        var claimSubject = new ClaimAutomationArgument(Guid.Parse(subject), DateTimeOffset.UtcNow);
+
+        try
+        {
+            claim = await claimAutomationRepository.AddClaimSubject(claimSubject);
+            var claimSubjectDto = new ClaimAutomationArgumentDto(claim.CreatedAt);
+
+            return CreatedAtAction(nameof(GetClaimAutomation), null, claimSubjectDto);
+        }
+        catch (DbUpdateException)
+        {
+            var claimSubjectDto = new ClaimAutomationArgumentDto(claimSubject.CreatedAt);
```

Comment:
Should refetch from the database here to return exact information for claim Automation Argument

Reply:
fixed